### PR TITLE
add GCS Storage Backend

### PR DESCRIPTION
**Pull request by rpahli**
this PR adds support for GCS as storage backend

### CODE REVIEW

**Comment by schabdo** [comment]
Glad to see first extension for GCP joining hawkBit. This is the last missing piece to run hawkBit on all major public clouds. So thanks a lot for your incredible valuable contribution! Heads up, we'll need some time to merge it to master: The `google-cloud-storage` dependency is introducing a bunch of new transitive dependency. It will take a bit to get them through the Eclipse IP process.

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-extension-artifact-repository-gcs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<!--
+
+    Copyright (c) 2015 Bosch Software Innovations GmbH and others.
```
Comment (schabdo) [inline hawkbit-extension-artifact-repository-gcs/pom.xml]: Correct me when I'm wrong but I am not sure if you're a fellow of `Bosch Software Innovations GmbH`. 
If not please change the title to give proper credit to either `Copyright (c) 2019 Rico Pahlisch and others` or `Copyright (c) 2019 <YOUR COMPANY> and others`. Depending on whether or not your employment contract stipulates that intellectual property developed by you is the property of your company. 

Don't forget to add the new header template [here](https://github.com/eclipse/hawkbit-extensions/tree/master/licenses).

**Comment by schabdo** [comment]
Thanks a lot for your patience @rpahli! We're very close to get this PR merged. From 23 dependencies in question 21 were successfully scanned and approved by the Eclipse IP-team. So only 2 dependencies to go. We'll keep you posted about the results. 

In the meanwhile may I ask you for the last time to update the PR and solve the merge conflicts? I think it's just new M4 parent-version and IntelliJ file exclude

**Comment by schabdo** [comment]
Thanks a lot @rpahli! We got another dependency cleared so only one left. I think on `master` a new method was introduced a few weeks ago. So `exists(String sha1)` should be `existsByTenantAndSha1()` now. This should do the trick to get build running again:

```
@Override
public boolean existsByTenantAndSha1(@NotEmpty final String tenant, @NotEmpty final String sha1Hash) {
   final Blob blob = gcsStorage.get(gcsProperties.getBucketName(), objectKey(tenant, sha1Hash));
   if (blob == null) {
        return false;
   }
   return blob.exists();
}
```

**Comment by rpahli** [comment]
Thanks for the feedback. Let's wait for the last dependency. :-)

**Comment by schabdo** [comment]
Awesome 👍

**Pull request merged**